File: 0383-ransom-note/0383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        vector<int>freq(26);
        for(auto it:magazine)
            freq[it-'a']++;
        for(auto it:ransomNote){
            if(!freq[it-'a'])
                return false;
            freq[it-'a']--;
        }
        return true;
    }
};

File: 0383-ransom-note/0383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        magazine = "".join(sorted(magazine))
        ransomNote = "".join(sorted(ransomNote))
        i = j = 0
        n, m = len(ransomNote), len(magazine)
        for i in range(m):
            if j < n and ransomNote[j] == magazine[i]:
                j += 1
        return bool(j == n)
